kong/plugins/openapi-doc/handle_api_doc.lua

```diff
@@ -118,6 +119,9 @@ local function process_paths(conf, api, doc, paths)
     if table_contains(tags_to_leave, tags_to_remove[i]) then
       tags_to_remove[i] = nil
     end
+    if api_prefix and table_contains(tags_to_leave, api_prefix .. tags_to_remove[i]) then
```

Comment:
This covers another bug: if the tag was in the path that should be added and the prefix was available the tag occurred with this prefix in the tags_to_leave array, so when checking it with the raw value from tags_to_remove (without prefix) it was always chosen to be removed.